atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Namespace.java

```diff
@@ -18,12 +18,15 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.benmanes.caffeine.cache.Interner;
 import com.palantir.logsafe.Preconditions;
 import java.util.Objects;
 import java.util.regex.Pattern;
 import org.apache.commons.lang3.Validate;
 
 public final class Namespace {
+    private static final Interner<String> names = Interner.newStrongInterner();
```

Comment:
fair enough, I guess there are fewer of these floating around and/or it's extremely unlikely for these to be transient